c/diff_ci.sh

```diff
@@ -26,5 +26,11 @@ dirs=`echo $relevant_diff | xargs dirname | cut -d/ -f2- | sort | uniq`
 printf "The CI commands will run in the following directories \n$dirs \n"
 for d in $dirs
 do
-  $cmdToExecute $d
+  if [ ! -f "$d/Makefile" ]; then
+    parent=`dirname $d`
+    # Execute the make only when the directory is not already present.
+    [[ "${dirs[@]}" =~ $parent ]] || $cmdToExecute $parent
```

Comment:
Won't this still break if I have two layers of directories without a Makefile, e.g.:

```
folder1
  |--Makefile
  |--folder2
       |--folder3
             |--someFileHereChanges
```

Won't this then still try to run Make inside of `folder2`?

Reply:
done